src/NUnitFramework/tests/Constraints/EqualConstraintTests.cs

```diff
@@ -180,6 +180,79 @@ public void ErrorIfTicksPrecedesWithin()
         #region DateTimeOffsetEquality
 
 #if !NETCF
+
+        public class DateTimeOffsetShouldBeSame
+        {
+      
+            [Datapoints]
+            public static readonly DateTimeOffset[] sameDateTimeOffsets = 
+                {
+                    new DateTimeOffset(new DateTime(2014, 1, 30, 12, 34, 56), new TimeSpan(6, 15, 0)), 
+                    new DateTimeOffset(new DateTime(2014, 1, 30, 9, 19, 56), new TimeSpan(3, 0, 0)),
+                    new DateTimeOffset(new DateTime(2014, 1, 30, 9, 19, 56), new TimeSpan(3, 1, 0)),
+                    new DateTimeOffset(new DateTime(2014, 1, 30, 9, 19, 55), new TimeSpan(3, 0, 0)),
+                    new DateTimeOffset(new DateTime(2014, 1, 30, 9, 19, 55), new TimeSpan(3, 50, 0))
+                };
+
+            [Theory]
+            public void PositiveEqualityTest(DateTimeOffset value1, DateTimeOffset value2)
+            {
+                Assume.That(value1 == value2);
+
+                Assert.That(value1, Is.EqualTo(value2));
+            }
+
+            [Theory]
+            public void NegativeEqualityTest(DateTimeOffset value1, DateTimeOffset value2)
+            {
+                Assume.That(value1 != value2);
+
+                Assert.That(value1, Is.Not.EqualTo(value2));
+            }
+
+            [Theory]
+            public void PositiveEqualityTestWithTolerance(DateTimeOffset value1, DateTimeOffset value2)
+            {
+                Assume.That((value1 - value2).Duration() <= new TimeSpan(0, 1, 0));
```

Comment:
In that case I'm ok work it. Anyone else?
On Apr 27, 2015 11:45 PM, "Vladimir Kloz" <notifications@github.com> wrote:

> In src/NUnitFramework/tests/Constraints/EqualConstraintTests.cs
> <https://github.com/nunit/nunit/pull/618#discussion_r29217809>:
>
> > +
> > +                Assert.That(value1, Is.EqualTo(value2));
> > +            }
> > +
> > +            [Theory]
> > +            public void NegativeEqualityTest(DateTimeOffset value1, DateTimeOffset value2)
> > +            {
> > +                Assume.That(value1 != value2);
> > +
> > +                Assert.That(value1, Is.Not.EqualTo(value2));
> > +            }
> > +
> > +            [Theory]
> > +            public void PositiveEqualityTestWithTolerance(DateTimeOffset value1, DateTimeOffset value2)
> > +            {
> > +                Assume.That((value1 - value2).Duration() <= new TimeSpan(0, 1, 0));
>
> According to MSDN
> <https://msdn.microsoft.com/en-us/library/system.timespan.duration%28v=vs.110%29.aspx>
> the Duration() method should return absolute value already so we should be
> good. Btw. I took that from original implementation of Tolerance support
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nunit/nunit/pull/618/files#r29217809>.
>